shuriken/lib/common/shurikenstream.cpp

```diff
@@ -102,6 +102,30 @@ std::string ShurikenStream::read_ansii_string(std::int64_t offset) {
     return new_str;
 }
 
+std::wstring ShurikenStream::read_unicode_string(std::int64_t offset) {
+    std::wstring new_str = L"";
+    std::uint16_t character = 0;
+    std::uint64_t utf16_size;
+
+    auto uint16_s = sizeof(std::uint16_t);
+    // save current offset
+    auto curr_offset = input_file.tellg();
+
+    // set the offset to the given offset
+    input_file.seekg(static_cast<std::streampos>(offset));
+
+    utf16_size = read_uleb128();
+
+    while (utf16_size-- > 0) {
+        input_file.read(reinterpret_cast<char*>(&character), uint16_s);
+        new_str += static_cast<wchar_t>(character);
```

Comment:
wchar_t on linux is 32bit and on windows its 16bit, maybe you want to move away from wchar_t and std::wstring usage and switch to std::u16string and char16_t (if dex files use 16bit unicode?) (-:

Reply:
I didn't know much about the `wchar_t`, yes, I can move to `std::u16string` and `char16_t`, I'm trying to figure out how to add all of this into the current disassembler to support both types of chars...

Reply:
Once I add this, I plan to fix headers and so on :_)